mlflow/pyfunc/scoring_server/client.py

```diff
@@ -0,0 +1,71 @@
+import requests
+import time
+from mlflow.pyfunc import scoring_server
+import subprocess
+import os
+import sys
+import signal
+import json
+import atexit
+import warnings
+
+
+class ScoringServerClient:
+    def __init__(self, host, port):
+        self.url_prefix = f"http://{host}:{port}"
+
+    def ping(self):
+        ping_status = requests.get(url=self.url_prefix + "/ping")
+        if ping_status.status_code != 200:
+            raise Exception(f"ping failed (error code {ping_status.status_code})")
+
+    def wait_server_ready(self, timeout=30):
+        begin_time = time.time()
+        while True:
+            time.sleep(0.3)
+            try:
+                self.ping()
+                return
+            except Exception:
+                pass
+            if time.time() - begin_time > timeout:
+                break
+        raise RuntimeError("Wait scoring server ready timeout.")
+
+    def invoke(self, data):
+        """
+        Invoke inference on input data. The input data must be pandas dataframe or json instance.
+        """
+
+        import pandas as pd
+
+        if isinstance(data, pd.DataFrame):
+            content_type = scoring_server.CONTENT_TYPE_JSON_RECORDS_ORIENTED
```

Comment:
Can we use the `split` format by default, since the records format might reorder prediction columns (dictionaries don't have a well-defined ordering)?

Reply:
Also filed https://github.com/mlflow/mlflow/pull/5522 to fix this for the sagemaker deployment plugin's `predict()` method. Can you review?

Reply:
The `application/json` content type should be fine for all Pandas DataFrame orientations because the scoring server uses the data type of the payload (dict or list) to determine whether to parse the Pandas DataFrame from records or split format. See https://github.com/mlflow/mlflow/blob/01182994ec31966dbe8528603d9e1a0ed2188234/mlflow/pyfunc/scoring_server/__init__.py#L93-L105.

Including this special case for added safety seems fine though :)

Reply:
Good catch!

Starting from python 3.6, the builtin dict preserve key order, but version lower than py3.6 it might cause column order changed.

Reply:
I saw "predictions_to_json" also use  pandas_orient="records", we need to also change it to split orient

Reply:
Change predictions_to_json format will change the returned value format, it is a breaking change, let's discuss it in follow-up task.
Most cases prediction result is one column data so it might be a minor issue.